src/solc_0.8/claims/MultiGiveaway/MultiGiveaway.sol

```diff
@@ -11,18 +11,29 @@ import {ERC2771Handler} from "../../common/BaseWithStorage/ERC2771Handler.sol";
 /// @notice This contract manages claims for multiple token types
 /// @dev The contract implements ERC2771 to ensure that users do not pay gas
 contract MultiGiveaway is AccessControl, ClaimERC1155ERC721ERC20, ERC2771Handler, Pausable {
+    /// @dev `bytes4(keccak256("onERC1155Received(address,address,uint256,uint256,bytes)"))`
     bytes4 private constant ERC1155_RECEIVED = 0xf23a6e61;
+
+    /// @dev `bytes4(keccak256("onERC1155BatchReceived(address,address,uint256[],uint256[],bytes)"))`
     bytes4 private constant ERC1155_BATCH_RECEIVED = 0xbc197c81;
+
+    /// @dev `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
     bytes4 internal constant ERC721_RECEIVED = 0x150b7a02;
+
+    /// @dev `bytes4(keccak256("onERC721BatchReceived(address,address,uint256[],bytes)"))`
     bytes4 internal constant ERC721_BATCH_RECEIVED = 0x4b808c46;
 
     mapping(address => mapping(bytes32 => bool)) public claimed;
-    mapping(bytes32 => uint256) internal _expiryTime;
+    mapping(bytes32 => uint256) private _expiryTime;
 
     event NewGiveaway(bytes32 merkleRoot, uint256 expiryTime);
-    event NewTrustedForwarder(address trustedForwarder);
+    event NewTrustedForwarder(address indexed trustedForwarder);
 
+    /// @notice Constructor with the admin & trusted forwarder
+    /// @param admin Admin of the contract
+    /// @param trustedForwarder Trusted forwarder for ERC2771
     constructor(address admin, address trustedForwarder) {
+        require(admin != address(0), "MULTIGIVEAWAY_INVALID_TO_ZERO_ADDRESS");
```

Comment:
We need to add a zeroAddress check for trustedForwarder too?

Reply:
QSP-3
trusted forwarder can be change after while if there's no admin => stucked